packages/date/typings/DateField.d.ts

```diff
@@ -3,6 +3,9 @@ import { DateProps } from './Date';
 
 export interface DateFieldProps extends DateProps {
   label?: React.ReactNode;
+  labelClass?: string;
+  labelHidden?: boolean;
+  labelAttrs?: object;
 }
```

Comment:
I see some of the old type definitions have `labelAttrs` set to `HTMLAtrributes` but do we have to restrict it like that?

Reply:
yeah its a stronger type than object. `React.HTMLAttributes<HTMLLabelElement>`. If someone passes something invalid in if they were writing typescript react will render and it will throw an error in the console even though the typescript definition allows it.

Reply:
In reality we should be checking if it has a custom tag and if so infer that type and use its props definition for the `labelAttrs`

Reply:
I was thinking that without the restriction you could pass props to the reactstrap `<Label />` that aren't necessarily attributes, but I guess we're trying to be restrictive for more control.

Reply:
such as `check`?